### PR TITLE
2 get topics

### PR DESCRIPTION
**Pull request by gilaineyo**
First endpoint /api/topics with error handling and testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+describe('/api/topics', () => {
+    test('GET 200 - responds with array of topic objects, with slug and description properties', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) => {
+            const { topics } = body
+            expect(Array.isArray(topics)).toBe(true)
+            expect(topics).toHaveLength(3)
```
Comment (johnhutcheon) [inline __tests__/app.test.js:18]: Nice work checking the length before a forEach 🙌

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            topics.forEach((topic) => {
+                expect(topic).toMatchObject({
+                    slug: expect.any(String),
+                    description: expect.any(String)
+                })
+            })
+        })
+    })
+    test('Error 404 - path not found', () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js:30]: We can pop this test in a separate describe block, as it's going to be relevant to the whole server and not just api/topics 👍

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+exports.getTopics = (req, res, next) => {
+    selectTopics()
+    .then((result) => {
+        res.status(200).send({ topics: result })
+    })
+}
+
+exports.handleBadPaths = (req, res) => {
+    res.status(404).send({ msg: 'Path not found' })
+}
```
Comment (johnhutcheon) [inline controllers/app.controllers.js]: I'd stick this controller in the error handling file 👍

**Review by johnhutcheon — APPROVED** [review]
Hey Gilaine, really nice work! Take a look at my comments, just some minor feedback for re-organising a couple of things, but all good to merge! 😄

File: __tests__/app.test.js (hunk 2 above)
Comment (gilaineyo) [reply __tests__/app.test.js:30]: Completed

File: controllers/app.controllers.js (hunk 3 above)
Comment (gilaineyo) [reply controllers/app.controllers.js]: Completed

**Pull request merged**